src/kestrel/session.py

```diff
@@ -418,12 +431,23 @@ def do_complete(self, code, cursor_pos):
                     elif token.startswith("STIXPATH"):
                         # TODO: figure out the varname and get its attrs
                         continue
+                    elif token.startswith("STIXPATTERNBODY"):
+                        # TODO: figure out how to complete STIX patterns
+                        continue
                     elif token == "RELATION":
-                        tmp.extend(all_relations)
+                        if last_word:
+                            tmp.extend(get_entity_types())
+                        else:
+                            tmp.extend(all_relations)
                     elif token == "REVERSED":
                         tmp.append("BY")
-                        varnames = self.get_variable_names()
-                        if last_word not in varnames:
+                        prev_word = words[-2] if len(words) >= 2 else ""
+                        _logger.debug("prev_word = %s", prev_word)
+                        if prev_word in all_relations:
```

Comment:
I think we can simplify `elif token == "REVERSED":` for only the case of `FIND`. We use keyword `"by"i` in `GROUP` and `SORT`, but not under `REVERSED`. I like the way we currently do the syntax, which differentiates generic `"by"i` from the special use of it as `REVERSED`. My guess is we do not need to handle those cases here.

Reply:
When `token == "REVERSED"` we don't know if it's a `FIND`, `GROUP`, or `SORT`.  That's why I look back to the previous word and check if it's a relation.

Reply:
I thought we only have `REVERSED` in `FIND`, `"by"i` in `GROUP` and `SORT` are not under `REVERSED` in the Lark syntax.

Reply:
Surprisingly, that's not the case.  You can see that when running the test cases.

Reply:
For example, the test case `grps = GROUP conns ` hits the `REVERSED` case, even though we use the literal `"by"i` in the rule for `group`.

Reply:
Interesting. Just did a Lark test, and this looks like a Lark logic that tries to use a terminal if exist. Also found a possible fix to tell Lark to differentiate them. Let me have a quick try to simplify it.